src/app/action-items/action-items.component.ts

```diff
@@ -0,0 +1,52 @@
+import { TasksService } from './../tasks.service';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'action-items',
+  templateUrl: './action-items.component.html',
+  styleUrls: ['./action-items.component.css']
+})
+export class ActionItemsComponent implements OnInit, OnDestroy {
+  color = '';
+  loading = true;
+  displayedColumns: string[] = ['title', 'type', 'completed', 'dueDate', 'daysLeft'];
+  dataSource;
+  dataSourceFetchedAgain;
+  private subscription: Subscription;
+
+  constructor(private tasksService: TasksService) {
+    this.subscription = this.tasksService.getAllTasks().subscribe(tasks => {
```

Comment:
Please move that to `ngOnInit` lifecycle method - https://blog.angularindepth.com/the-essential-difference-between-constructor-and-ngoninit-in-angular-c9930c209a42